presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergDistributedQueries.java

```diff
@@ -0,0 +1,244 @@
+        assertEquals(totalCount - countPart1F - countPart2O - countPartOther, newTotalCount);
+        assertQuerySucceeds("DROP TABLE test_partitioned_drop");
+
+        // Support delete with filters on non-identity partition column
+        assertUpdate("CREATE TABLE test_partitioned_drop WITH (partitioning = ARRAY['bucket(orderkey, 2)', 'linenumber', 'linestatus']) as select * from lineitem", totalCount);
+        long countOrder1 = (long) getQueryRunner().execute("SELECT count(*) FROM test_partitioned_drop where orderkey = 1").getOnlyValue();
+        assertUpdate("DELETE FROM test_partitioned_drop WHERE orderkey = 1", countOrder1);
+        long countPartKey100 = (long) getQueryRunner().execute("SELECT count(*) FROM test_partitioned_drop where partkey > 100").getOnlyValue();
+        assertUpdate("DELETE FROM test_partitioned_drop WHERE partkey > 100", countPartKey100);
+        long countLine1Order1 = (long) getQueryRunner().execute("SELECT count(*) FROM test_partitioned_drop where linenumber = 1 and orderkey = 1").getOnlyValue();
+        assertUpdate("DELETE FROM test_partitioned_drop WHERE linenumber = 1 and orderkey = 1", countLine1Order1);
+
+        // Do not allow delete data at specified snapshot
+        String errorMessage2 = "This connector do not allow delete data at specified snapshot";
+        List<Long> snapshots = getQueryRunner().execute("SELECT snapshot_id FROM \"test_partitioned_drop$snapshots\"").getOnlyColumnAsSet()
+                .stream().map(Long.class::cast).collect(Collectors.toList());
+        for (long snapshot : snapshots) {
+            assertQueryFails("DELETE FROM \"test_partitioned_drop@" + snapshot + "\" WHERE linenumber = 1", errorMessage2);
+        }
+
+        assertQuerySucceeds("DROP TABLE test_partitioned_drop");
+    }
+
+    @Override
+    public void testShowColumns()
+    {
+        MaterializedResult actual = computeActual("SHOW COLUMNS FROM orders");
+
+        MaterializedResult expectedParametrizedVarchar = resultBuilder(getSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+                .row("orderkey", "bigint", "", "")
+                .row("custkey", "bigint", "", "")
+                .row("orderstatus", "varchar", "", "")
+                .row("totalprice", "double", "", "")
+                .row("orderdate", "date", "", "")
+                .row("orderpriority", "varchar", "", "")
+                .row("clerk", "varchar", "", "")
+                .row("shippriority", "integer", "", "")
+                .row("comment", "varchar", "", "")
+                .build();
+
+        Assert.assertEquals(actual, expectedParametrizedVarchar);
+    }
+
+    /**
+     * Increased the optimizer timeout from 15000ms to 25000ms
+     */
+    @Override
+    public void testLargeIn()
+    {
+        String longValues = range(0, 5000)
+                .mapToObj(Integer::toString)
+                .collect(joining(", "));
+        Session session = Session.builder(getSession())
```

Comment:
Is there a reason these all need to be in the same session? otherwise these could be independent more unitary test methods

Reply:
I think if we can make sure that the `transactionId` is not shared, as `Session.build(session)` has checked already, then we can consider the sessions built on this way as stateless ones. However, I'm not so sure, will check out this.

Reply:
Hi @elharo, sorry for not having much time recently, I will spend some time later to  organize in detail whether there could be any issues of mutual influence between the sessions constructed by `Session.build(session)`. Once done, I will still reply here.